File: client/user-web/src/model/global/dto/responseDto.ts
import type ResponseErrorDto from "@/model/global/dto/responseErrorDto"

/**
 * 공통 ResponseDto
 * 
 * T: data?
 * ResponseErrorDto: error?
 */
export default interface ResponseDto<T> {
    timestamp: string
    dto?: T
    error?: ResponseErrorDto
}

File: server/src/main/java/com/haloomin/goal/global/response/dto/ResponseDto.java
package com.haloomin.goal.global.response.dto;

import jakarta.annotation.Nullable;
import lombok.Builder;

import java.time.LocalDateTime;

public record ResponseDto<T>(
        LocalDateTime timestamp,
        @Nullable
        T dto,
        @Nullable
        ResponseDtoError error
) {

    @Builder
    public ResponseDto {
        if (timestamp == null) {
            timestamp = LocalDateTime.now();
        }
    }
}
